mmedit/datasets/transforms/crop.py

```diff
@@ -916,3 +921,107 @@ def __repr__(self):
+                 finesize=256):
+
+        cfg = get_config(config_file, pretrained=True)
+        with DefaultScope.overwrite_default_scope('mmdet'):
+            self.predictor = init_detector(cfg, cfg.model_path)
+
+        self.key = key
+        self.box_num_upbound = box_num_upbound
+        self.final_size = finesize
+
+    def transform(self, results: dict) -> dict:
+        """The transform function of InstanceCrop.
+
+        Args:
+            results (dict): A dict containing the necessary information and
+                data for Conversion
+
+        Returns:
+            results (dict): A dict containing the processed data
+                and information.
+        """
+        # get consistent box prediction based on L channel
+        full_img = results['img']
+        full_img_size = results['ori_img_shape'][:-1][::-1]
+        pred_bbox, pred_scores = self.predict_bbox(full_img)
+
+        if self.box_num_upbound > 0 and pred_bbox.shape[
+                0] > self.box_num_upbound:
+            index_mask = np.argsort(pred_scores, axis=0)
+            index_mask = index_mask[pred_scores.shape[0] -
+                                    self.box_num_upbound:pred_scores.shape[0]]
+            pred_bbox = pred_bbox[index_mask]
+
+        # get cropped images and box info
+        cropped_img_list = []
+        index_list = range(len(pred_bbox))
+        box_info, box_info_2x, box_info_4x, box_info_8x = np.zeros(
+            (4, len(index_list), 6))
+        for i in index_list:
+            startx, starty, endx, endy = pred_bbox[i]
+            cropped_img = full_img[starty:endy, startx:endx, :]
+            cropped_img_list.append(cropped_img)
+            box_info[i] = np.array(
+                get_box_info(pred_bbox[i], full_img_size, self.final_size))
+            box_info_2x[i] = np.array(
+                get_box_info(pred_bbox[i], full_img_size,
+                             self.final_size // 2))
+            box_info_4x[i] = np.array(
+                get_box_info(pred_bbox[i], full_img_size,
+                             self.final_size // 4))
+            box_info_8x[i] = np.array(
+                get_box_info(pred_bbox[i], full_img_size,
+                             self.final_size // 8))
+
+        # update results
+        if len(pred_bbox) > 0:
+            results['cropped_img'] = cropped_img_list
+            results['box_info'] = torch.from_numpy(box_info).type(torch.long)
+            results['box_info_2x'] = torch.from_numpy(box_info_2x).type(
+                torch.long)
+            results['box_info_4x'] = torch.from_numpy(box_info_4x).type(
+                torch.long)
+            results['box_info_8x'] = torch.from_numpy(box_info_8x).type(
+                torch.long)
+            results['empty_box'] = False
+        else:
+            results['empty_box'] = True
+        return results
+
+    def predict_bbox(self, image):
+        lab_image = cv.cvtColor(image, cv.COLOR_BGR2LAB)
```

Comment:
Type lint in the function signature is missing. (Maybe can convert to a **good first issue**)

Reply:
Agree.